src/components/product.js

```diff
@@ -461,7 +477,7 @@ export default class Product extends Component {
    * @return {String}
    */
   get onlineStoreURL() {
-    const identifier = this.handle ? this.handle : this.id;
+    const identifier = this.handle ? this.handle : atob(this.id).split('/').pop();
```

Comment:
Let's add an issue to the storefront API feature board @swalkinshaw created about getting this added to the API so it's not lost.